src/components/FlippableHandler.js

```diff
@@ -32,21 +33,56 @@ class FlippableHandler {
                         window.setTimeout(() => $oldEl.classList.remove('display-flex'), 300);
                         window.setTimeout(() => {
                             $rotationContainer.classList.toggle('flipped', $newEl.classList.contains(classname));
+                            FlippableHandler._updateContainerHeight($newEl);
                         }, 0);
+                    } else {
+                        FlippableHandler._updateContainerHeight($newEl || undefined);
                     }
                 } else if (newHash) {
                     const $newEl = document.querySelector(newHash);
                     if ($newEl && $newEl.classList.contains(classname)) {
                         $rotationContainer.classList.add('disable-transition');
+                        FlippableHandler._updateContainerHeight($newEl);
                         window.setTimeout(() => {
                             $rotationContainer.classList.toggle('flipped', $newEl.classList.contains(classname));
                             window.setTimeout(() => $rotationContainer.classList.remove('disable-transition'), 10);
                         }, 0);
+                    } else {
+                        window.setTimeout(() => {
+                            $rotationContainer.classList.add('disable-transition');
+                            FlippableHandler._updateContainerHeight($newEl || undefined);
+                            window.setTimeout(() => {
+                                $rotationContainer.classList.remove('disable-transition');
+                            }, 0);
+                        }, 0);
                     }
                 }
             });
             FlippableHandler.flippableHandlerInitialised = true;
         }
     }
+
+    /**
+     * Update the height of the #rotation-container element to match its content.
+     * The default behavior is to look for every visible `.page` element,
+     * and, if there is multiple, take the height of the higher.
+     * @param {Element} [$enforcedElement] - Enforce which element the function is taking the height from.
+     *  Must be a child of `#rotation-container`
+     */
+    static _updateContainerHeight($enforcedElement) {
+        /** @type {HTMLElement} */
+        const $rotationContainer = (document.getElementById('rotation-container'));
+        if ($enforcedElement && $rotationContainer.contains($enforcedElement)) {
+            $rotationContainer.style.height = `${$enforcedElement.clientHeight}px`;
+        } else {
+            /** @type {Array<HTMLElement>} */
+            const $pages = Array.from($rotationContainer.querySelectorAll('.page'));
+            if ($pages && $pages.length > 0) {
+                const heights = $pages.map($el => ($el.offsetParent ? $el.clientHeight : 0));
```

Comment:
The check here isn't necessary as `clientHeight` is automatically 0 if the element is not attached to the DOM.